datalad_next/utils/__init__.py

```diff
@@ -28,6 +40,12 @@
 
 from datalad_next.credman import CredentialManager
 from .log import log_progress
+from .shell_connection import (
+    shell_connection,
```

Comment:
I wonder if it could simply be a `shell`. A meaningful use case is also a persistent shell subprocess running on a local machine without an intermediate network transport. Seems to work fine:

```py
In [3]: with shell_connection(['bash']) as shell:
   ...:     r = shell.execute_command(b'uname')
   ...:     print(list(r))
   ...:     print(r.returncode)
   ...: 
[b'Linux\n']
0
```

Reply:
I like `shell`, but it is quite generic and I was not sure whether we have another concept that is "closer" to `shell`. I don't mind renaming it.

Reply:
Renamed `datalad_next.utils.shell_connection.shell_connection` to `datalad_next.utils.shell.shell` in b42e435b4e711013bc1ae385e7ad18fdf3593b26